src/connection/usb_connection.rs

```diff
@@ -49,6 +49,30 @@ impl UsbConnectionInfo {
             timeout: Duration::from_millis(5000),
         }
     }
+
+    /// Find a connected printer and return its connection info.
+    ///
+    /// Returns the correct USB parameters (vendor ID, product ID, endpoints, etc.)
+    /// for the first supported printer that's found. This is the recommended way to
+    /// create connection info if you do not care which printer model you have.
+    ///
+    /// # Errors
+    /// Returns an error if any USB operations fail
+    pub fn discover() -> Result<Option<Self>, UsbError> {
+        let context = Context::new()?;
+        let devices = context.devices()?;
+
+        for device in devices.iter() {
+            let descriptor = device.device_descriptor()?;
+            if descriptor.vendor_id() == 0x04f9 {
```

Comment:
This Brother VID is used in 2 different places now, might be a good opportunity to put it as a constant in a single place. Not sure where it best fits  though.